libs/langgraph-store-mongodb/langgraph/store/mongodb/base.py

```diff
@@ -0,0 +1,692 @@
+                    self.search(
+                        op.namespace_prefix,
+                        query=op.query,
+                        filter=op.filter,
+                        limit=op.limit,
+                        offset=op.offset,
+                        refresh_ttl=op.refresh_ttl,
+                    )
+                )
+
+            elif isinstance(op, ListNamespacesOp):
+                prefix = None
+                suffix = None
+                if op.match_conditions:
+                    for cond in op.match_conditions:
+                        if cond.match_type == "prefix":
+                            prefix = cond.path
+                        elif cond.match_type == "suffix":
+                            suffix = cond.path
+                        else:
+                            raise ValueError(
+                                f"Match type {cond.match_type} must be prefix or suffix."
+                            )
+                results.append(
+                    self.list_namespaces(
+                        prefix=prefix,
+                        suffix=suffix,
+                        max_depth=op.max_depth,
+                        limit=op.limit,
+                        offset=op.offset,
+                    )
+                )
+        # Apply puts and deletes in bulk
+        # Extract texts to embed for each op
+        if self.index_config:
+            texts = self._extract_texts(list(dedupped_putops.values()))
+            vectors = self.embeddings.embed_documents(texts)
+            v = 0
+        for op in dedupped_putops.values():
+            if op.value is None:
+                # mark the item for deletion.
+                writes.append(
+                    DeleteOne(filter={"namespace": list(op.namespace), "key": op.key})
+                )
+            else:
+                # Add or Upsert the value
+                to_set = {
+                    "value": op.value,
+                    "updated_at": datetime.now(tz=timezone.utc),
+                }
+                if self.index_config:
+                    to_set[self._embedding_key] = vectors[v]
+                    to_set["namespace_prefix"] = self._denormalize_path(op.namespace)
+                    v += 1
+
+                writes.append(
+                    UpdateOne(
+                        filter={"namespace": list(op.namespace), "key": op.key},
+                        update={
+                            "$set": to_set,
+                            "$setOnInsert": {
+                                "created_at": datetime.now(tz=timezone.utc),
+                            },
+                        },
+                        upsert=True,
+                    )
+                )
+
+        if writes:
+            self.collection.bulk_write(writes)
+        return results
+
+    async def abatch(self, ops: Iterable[Op]) -> list[Result]:
+        """Execute multiple operations asynchronously in a single batch.
```

Comment:
Why is this the only async method?

Reply:
It is the only `@abstractmethod` of the base class.